pkg/Library/Fields/TextField.js

```diff
@@ -5,10 +5,10 @@
  * license that can be found in the LICENSE file.
  */
 ({
-update({value, connectedValue}, state) {
-  value = state.value = connectedValue ?? value;
-  return {value};
+update({value}, state) {
+  state.value = value;
```

Comment:
The only reason this was copied to `state` previously was because it was calculated.

You could remove `update` entirely in this case, and use `inputs.value` directly.

However, I recently wanted to bind two sources (one microphone transcript, the other a select box) as inputs for MultilineTextField. This led to some logic in `update` like this [MultilineTextField.js](https://github.com/project-oak/arcsjs-core/blob/main/pkg/Library/Fields/MultilineTextField.js#L9).

That's not great, but a more polished solution might be a pattern to consider for text fields.

Aside: the ability to do this kind of state logic in `update` is one of the primary advances of XenState from Polymer.

Reply:
updated!

i changed how this works with MultilineTextField, select and mic (and just realized, it is in another repo, will make a PR now): both the select and the mic `connect` to the MultilineTextField's value store.
this is something i wanted to discuss - nodegraph doesn't fully support (e.g. Editor doesn't visualize it well) using `connection` for only output bindings... otoh it makes sense in the text+select+mic use case. WDYT?

Reply:
corresponding labs pr: https://team-review.git.corp.google.com/c/arcs-team/arcsjs-labs/+/1678719